spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -50,6 +55,7 @@
 
   private final SparkSession spark;
   private final JavaSparkContext sparkContext;
+  private static final AtomicInteger jobCounter = new AtomicInteger();
```

Comment:
Since this is a static variable now, should we move it up and use `JOB_COUNTER` as the name?

I'd also add an empty line to separate static and instance variables like this:

```
private static final AtomicInteger JOB_COUNTER = new AtomicInteger();

private final SparkSession spark;
private final JavaSparkContext sparkContext;
```